lib/Rex/Commands/File.pm

```diff
@@ -357,6 +372,19 @@ sub file {
     }
 
     return \@ret;
+  } 
+  elsif ( my @exp = $file =~ m/^((?:\\.|[^\\\{])*)({(?:\\.|[^\\\*?\[])*})((?:\\.|[^\\\}])*)$/ ) {
+    if ( $exp[1] =~ /{[^},]*,[^}]*}/ ) {
+      my @ret;
+      # quote braces to keep macros
+      $exp[1] =~ s/({[^,}]*)}/\\$1\\}/g;
+  
+      for my $f ( glob $exp[1] ) {
+          push ( @ret, file( "$exp[0]$f$exp[2]", @options) );
+      }
+
+      return \@ret;
+    }
```

Comment:
I have a hard time understanding what this code branch does, especially the regex in the condition. Can you explain it, please? I probably miss a corner case you found, because I expected the solution would be simpler, something like:
```perl
my @files_after_glob = glob($file);

if ( scalar @files_after_glob > 1 ) {
    Rex::Logger::debug( "Files after glob: " . join ', ', @files_after_glob);
    my $ret = file( \@files_after_glob, @options );
    return $ret;
}
```